search-service/src/main/java/de/cxp/ocs/elasticsearch/Searcher.java

```diff
@@ -321,6 +332,36 @@ public SearchResult find(InternalSearchParams parameters) throws IOException {
 		return searchResult;
 	}
 
+	/**
+	 * For simple word filters the returned searchWords are used
+	 * For any special Querqy style filtering they are put into the parameters object
+	 * @param parameters
+	 * @param searchWords
+	 * @return
+	 */
+	private List<QueryStringTerm> handleFiltersOnFields(InternalSearchParams parameters, List<QueryStringTerm> searchWords) {
+		// Pull all QueryFilterTerm items into a list of its own
+		List<QueryFilterTerm> additionalQuerqyFilters = new ArrayList<>();
+		searchWords.stream().filter(searchWord -> searchWord instanceof QueryFilterTerm).forEach(
+			queryFilterTerm -> additionalQuerqyFilters.add((QueryFilterTerm)queryFilterTerm));
+		searchWords.removeAll(additionalQuerqyFilters);
+
+		// Generate the filters and add them
+		Map<String, String> filtersAsMap =
+			additionalQuerqyFilters.stream().collect(Collectors.toMap(QueryFilterTerm::getWord, QueryFilterTerm::getField));
```

Comment:
Shouldn't the key be the "Field" and the value be the "Word"? (I already changed that in the impl. example above)

Reply:
I had it at first, but actually had to inverse it

Reply:
I found the reason for it: The order of the QueryFilterTerm arguments was inverted. Fixed it.